### ISSUE TITLE
Error installing / we_megamenu compatibility

### ISSUE THREAD

**Issue by ugintl** [droptica/droopler_project#128]
PHP: 7.4. Composer 2. Node 13. I have just tried to install droopler 3.0. I am getting following error/notice.

![droopler](https://user-images.githubusercontent.com/12570802/151415216-77997b00-fa58-484a-9015-5d5849f48a76.jpeg)

**Comment by larsskj** [comment]
I get the same error. Tried PHP 7.4, 8.0, and 8.1.

**Comment by larsskj** [comment]
And just to help people searching for this error: The primary error message is _Undefined array key "d_custom_init"_.

**Comment by bem22** [comment]
I have exactly the same issue today

Composer 2.2.6
Gulp 2.3.0
NPM 7.5.2

**Comment by grzegorz-pietrzak-droptica** [comment]
I've added the ticket to our issue queue on drupal.org:
https://www.drupal.org/project/droopler/issues/3265785

**Comment by bem22** [comment]
This also appears later in the installation process 
![image](https://user-images.githubusercontent.com/29655042/155521816-b8f28215-3e5f-40f5-8c51-eb5852f0afb9.png)

**Comment by larsskj** [comment]
Yes, that happens for me as well: Whilst the first error really is a warning only and allows you to continue, this is a real error and a showstopper preventing a completed install.

**Comment by grzegorz-pietrzak-droptica** [comment]
This is caused by we_megamenu module, which throws deprecation warnings and breaks AJAX calls in the installer. The solution for now is either applying patch https://www.drupal.org/project/we_megamenu/issues/3239043 or changing PHP error reporting level to `E_ERROR` for the installation.

**Comment by bem22** [comment]
> This is caused by we_megamenu module, which throws deprecation warnings and breaks AJAX calls in the installer. The solution for now is either applying patch https://www.drupal.org/project/we_megamenu/issues/3239043 or changing PHP error reporting level to `E_ERROR` for the installation.

I will apply and confirm

**Comment by larsskj** [comment]
Just tried patching, but unfortunately, the error above persists.

Still cannot build a working site.

**Comment by grzegorz-pietrzak-droptica** [comment]
Are you sure the patch is working? I checked this one yesterday, and it worked

```json
"patches": {
    "drupal/we_megamenu": {
        "Fix deprecations": "https://www.drupal.org/files/issues/2022-02-24/we_megamenu_deprecated.patch"
    }
}
```

**Comment by larsskj** [comment]
Running `composer install` claims:

```
  - Installing drush/drush (10.6.2): Extracting archive
  - Applying patches for drupal/we_megamenu
    https://www.drupal.org/files/issues/2022-02-24/we_megamenu_deprecated.patch (Deprecated error/PHP 8.0 Drupal 9)

Package doctrine/reflection is abandoned, you should avoid using it. Use roave/better-reflection instead.
Package webmozart/path-util is abandoned, you should avoid using it. Use symfony/filesystem instead.
Generating autoload files
93 packages you are using are looking for funding.
```

and I have

```
"enable-patching": "true",
"patches": {
    "drupal/we_megamenu": {
        "Deprecated error/PHP 8.0 Drupal 9": "https://www.drupal.org/files/issues/2022-02-24/we_megamenu_deprecated.patch"
    }
}
```

in my `composer.json`.

**Comment by larsskj** [comment]
I have now tried setting it up four times: English as language, two times enabling `d_blog` and demo content, one time without any additional components at all, and one time enabling all possible components (except for `d_commerce`).

Every attempt invariably end up in a WSOD. And no information found in the Apache `error.log`.

**Comment by grzegorz-pietrzak-droptica** [comment]
Maybe you'll be able to install it by drush, this is the full command with all modules:

```bash
drush site-install droopler install_configure_form.enable_update_status_module=TRUE install_configure_form.enable_update_status_emails=NULL --db-url='mysql://drupal9:drupal9@database/drupal9' --account-name=admin --account-pass=123 --site-name=droopler -y --locale=en droopler_additional_modules_form.module_d_blog=1 droopler_additional_modules_form.module_d_product=1 droopler_additional_modules_form.init_content=1 droopler_additional_modules_form.documentation=1
```

**Comment by grzegorz-pietrzak-droptica** [comment]
I am not getting this issue, maybe it is tied to some PHP version?

**Comment by bem22** [comment]
I am getting this issue - still. @larsskj , any luck yet?

**Comment by larsskj** [comment]
Nope - no luck. Basic error is the same - I still get Undefined array key "d_custom_init".

Built using PHP 8.1.3 - executed in Docker using PHP 8.0.15.

And there's no subtheme, by the way.

**Comment by bem22** [comment]
Hello,

I was finally able to install the theme and demo content. What I was trying to do from the beggining was to use an official docker image from drupal. However, the one you pull with no other arguments (latest) also has some of the latest dependencies. @grzegorz-pietrzak-droptica suggested that PHP might play a key role here so I went all the way back to the final release of droopler (July 2021) and found a docker image that was suitable (9.2.0). Indeed, using the dependencies required for drupal 9.2.16 was the solution for the above problem. 

In the same (working) image I have the following dependencies:
- Composer 2.2.9
- PHP 7.4.28
- Apache 2.4.52
- NPM 8.5.0
- Node 16.14.2


Now one question remains - How compatible is the latest drupal release which has all the security fixes with php 7.4? 

Also there are some other dependencies issues pointed by npm install. I will open a new issue later this week.

Thanks and good luck

**Comment by grzegorz-pietrzak-droptica** [comment]
Please have a look at https://www.drupal.org/docs/system-requirements/php-requirements
Generally, it is about error logging, which breaks AJAX output. If you turn off PHP notices for the time of installation, it will all work fine.

**Comment by grzegorz-pietrzak-droptica** [comment]
Hello, on Friday we released 3.1.0-rc1 version with `we_megamenu` patched for full PHP 8.1 support.
